kornia/feature/affine_shape.py

```diff
@@ -0,0 +1,118 @@
+from typing import Tuple
+
+import torch
+import torch.nn as nn
+import math
+from kornia.filters import get_gaussian_kernel2d
+from kornia.filters import SpatialGradient
+from kornia.feature.laf import (ellipse_to_laf,
+                                get_laf_scale,
+                                raise_error_if_laf_is_not_valid,
+                                scale_laf, make_upright)
+from kornia.feature import extract_patches_from_pyramid
+
+
+class PatchAffineShapeEstimator(nn.Module):
+    def __init__(self, patch_size: int = 19, eps: float = 1e-10):
+        """Module, which estimates the second moment matrix of the patch gradients in order to determine the
+        affine shape of the local feature as in
+        A. Baumberg (2000). "Reliable feature matching across widely separated views".
```

Comment:
I think there's a way to generate a reference like in latex

Reply:
Thanks, I`ll check it and modify